clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -75,6 +82,8 @@
 
     /* max time in ms for the producer to wait for acknowledgement from server*/
     private final int requestTimeoutMs;
+    
+    private final Time time;
```

Comment:
@enothereska What do you think about my suggestion above? I think it's a little confusing to have NetworkClient.poll() both accept a `now` parameter and update using an internal `Time` reference.